src/proformellipticaltrainer.cpp

```diff
@@ -567,22 +568,24 @@ void proformellipticaltrainer::stateChanged(QLowEnergyService::ServiceState stat
 
         // ******************************************* virtual treadmill init *************************************
         QSettings settings;
-        if (!firstStateChanged && !virtualTreadmill && !virtualBike) {
-            bool virtual_device_enabled = settings.value(QZSettings::virtual_device_enabled, QZSettings::default_virtual_device_enabled).toBool();
+        if (!firstStateChanged && !VirtualDevice()) {
```

Comment:
Failure to use hasVirtualDevice(). This won't work as intended when using the virtual bike.